docs: modular dependency imports for langflow-base (#11250)
langflow-ai/langflow: docs/docs/Support/release-notes.mdx

Highlights of this release include the following changes.
For all changes, see the [Changelog](https://github.com/langflow-ai/langflow/releases).

### Breaking changes

> -   `langflow-base` dependency structure refactor

Reviewer: ```suggestion
-   `langflow-base` dependency structure refactored
```

    The `langflow-base` package has been refactored to use granular optional dependency groups. Many dependencies that were previously included in the `langflow-base` installation have been moved to optional extras.

>     If you installed Langflow with `uv pip install langflow`, this is not a breaking change. Installing `langflow` automatically installs `langflow-base[complete]`, which includes all optional dependencies and maintains the same functionality as before.

Reviewer: ```suggestion
    If you installed Langflow with `uv pip install langflow`, this isn't a breaking change. Installing `langflow` automatically installs `langflow-base[complete]`, which includes all optional dependencies and maintains the same functionality as before.
```

    If you installed Langflow with `uv pip install langflow-base` without specifying extra dependencies, **this is a breaking change**.
    Dependencies that were previously included by default are now available only through optional extras.
    If you install `langflow-base` directly, you will now only install the [core base dependencies](https://github.com/langflow-ai/langflow/blob/main/src/backend/base/pyproject.toml).

>     **Migration**: If you use `langflow-base` directly and need the full set of dependencies, install it with the `complete` extra:

Reviewer: ```suggestion
    If you installed `langflow-base`, there are two ways to resolve dependency errors that result from this breaking change:
    
    * If you need the full set of dependencies, you must install `langflow-base` with the `complete` extra:
```
